### ISSUE TITLE
[MongoSource] Error reading from MongoDB: Circular reference detected

### ISSUE THREAD

**Issue by DevBey** [mage-ai/mage-ai#4958]
### Mage version

0.9.67

### Describe the bug

[MongoSource] Error reading from MongoDB: Circular reference detected

while syncing mongodb to postgres using CDC streaming. 

it gives circular reference error. 

previously we were using airbyte for this, but want to shift to mage as it looks more light weight and fast. 

### To reproduce

1. go to data loader create connetion to mongodb
2. go to postgres and create connection to posgres.
3. insert a document in mongodb collection being synced 
4. errors occours.

### Expected behavior

sync data to postgres without any errors 

### Screenshots

![image](https://github.com/mage-ai/mage-ai/assets/85338857/c971c04d-354d-4f2d-9153-c4a47b8bf263)

### Operating system

- eks
- amd64
- ubuntu

**Comment by DevBey** [comment]
Hello @wangxiaoyou1993, 

but this is the only log we get for this, apologies but please let me know if I am missing anything 

INFO:cold_sea/2/20240418T113235:{"pipeline_run_id": 5, "pipeline_schedule_id": 2, "pipeline_uuid": "cold_sea", "hostname": "d89761f2784e", "level": "INFO", "message": "[MongoSource] Error reading from MongoDB: Circular reference detected", "timestamp": 1713463027.410202, "uuid": "cae3c7f1925a44a49a101c916d5f1a58"}

**Comment by wangxiaoyou1993** [comment]
@DevBey can you run the pipeline in the editor page and see there's stacktrace?

**Comment by DevBey** [comment]
apologies but how ?? 

![image](https://github.com/mage-ai/mage-ai/assets/85338857/07a2f96c-a7da-4104-9ee3-7688335c52c1)

**Comment by wangxiaoyou1993** [comment]
@DevBey you should have the button to execute the pipeline 
<img width="398" alt="image" src="https://github.com/mage-ai/mage-ai/assets/80284865/49c7bda4-69d1-4373-bf8e-6fe75bd6a603">

**Comment by DevBey** [comment]
yep we tried this same traceback, apart from this just logs of message received and message being inserted into postgres with the mongodb document. 

then this error, no other trackback. 
![image](https://github.com/mage-ai/mage-ai/assets/85338857/93ed9647-50ff-4c46-8696-8add78e12869)

**Comment by DevBey** [comment]
Hello @wangxiaoyou1993, 

could you please help with any update on this ??

**Comment by DevBey** [comment]
Hello @wangxiaoyou1993,

quick update on this, mongdb to postgres streaming does work in case we do some kind of transformation and then pass on it to postgres data exporter.

**Comment by wangxiaoyou1993** [comment]
> Hello @wangxiaoyou1993,
> 
> quick update on this, mongdb to postgres streaming does work in case we do some kind of transformation and then pass on it to postgres data exporter.

What transformation did you add?

**Comment by DevBey** [comment]
nothing much just few custom transforms that we do for data type issues like dates etc. 

we figured that mongodb issue would be eliminated if we instead of passing mongodb loader to postgres exporter directly just, do some kind of initial transform and then pass on the message to postgres exporter and it worked.

**Comment by wangxiaoyou1993** [comment]
> nothing much just few custom transforms that we do for data type issues like dates etc.
> 
> we figured that mongodb issue would be eliminated if we instead of passing mongodb loader to postgres exporter directly just, do some kind of initial transform and then pass on the message to postgres exporter and it worked.

Thanks for the info! Will test with the dates type.

**Comment by DevBey** [comment]
apologies but my hunch is that I don't think it's related to any particular datatype. it's a generaly issue when fetching data from mongodb and dumping into Postgres. it works MongoDB to MongoDB

**Comment by harelo-cf** [comment]
@DevBey Did you figure out a workaround or is it still no go with Mage?

**Comment by DevBey** [comment]
yep we fetch the data and then do a transformation step in between, if we do that then we don't get such errors. 

ps:- we are already using tens of such pipeline for last few months now, so no worries.

**Comment by harelo-cf** [comment]
@DevBey I tried to do exactly what you suggested but it didn't work, can you share an example please?

I have tried to just return a List[Dic] and each dic is just {'doc_id': uuid4(), 'doc_value': ...}, it didn't work for me..
